shared/tasks/browser.mjs

```diff
@@ -82,27 +91,31 @@ export async function screenshotComponent(browser, componentName, options) {
   // Navigate to component
   const page = await goToComponent(browser, componentName, options)
 
-  // Add optional example to screenshot name
-  const screenshotName = options?.exampleName
-    ? `${componentName} (${options.exampleName})`
-    : componentName
+  const screenshotName = `${componentName} (${options.exampleName})`
 
-  // Screenshot preview page (with JavaScript)
-  await percySnapshot(page, `js: ${screenshotName}`, snapshotOptions)
-
-  // Check for "JavaScript enabled" components
-  if (componentFiles.some(filterPath([`**/${componentName}.mjs`]))) {
-    await page.setJavaScriptEnabled(false)
-
-    // Screenshot preview page (without JavaScript)
-    await page.reload({ waitUntil: 'load' })
-    await percySnapshot(page, `no-js: ${screenshotName}`, snapshotOptions)
+  if (options.screenshot.variants?.noJs) {
+    await percySnapshotNoJs(page, screenshotName, snapshotOptions)
   }
 
+  await percySnapshot(page, screenshotName, snapshotOptions)
+
   // Close page
   return page.close()
 }
 
+/**
+ *
+ * @param {Page} page - Puppeteer page object
+ * @param {string} screenshotName - The name of the screenshot
+ * @param {SnapshotOptions} snapShotOptions - Percy snapshot options
+ */
+export async function percySnapshotNoJs(page, screenshotName, snapShotOptions) {
+  await page.setJavaScriptEnabled(false)
+  await page.reload({ waitUntil: 'load' })
+  screenshotName = `no-js: ${screenshotName}`
+  await percySnapshot(page, screenshotName, snapShotOptions)
```

Comment:
Whoops, To do: ensure that Page is fully reset before and after each variant function. Might need to move the Page creation step.